### PR TITLE
Add applicant related operations

### LINKED ISSUE #294: [Feature]: PeopleHR connector - support applicant details insertion
### Connector Name

module/peoplehr (People HR)

### Suggested feature

Request email: "[Choreo] People HR connector feature request"

We have a requirement to integrate [wso2.com](http://wso2.com/) with People HR.
This is to insert applicant details we received from [wso2.com](http://wso2.com/) to the vacancies we have in PeopleHR.

People HR API v3.1[1] supports;

- Create new applicant
- Upload new applicant document
- Check duplicate applicant

Refer: [2]

We need the above feature update for the current PeopleHR connector on Choreo V2.
Let me know if you have any questions.

[1][https://apidocs.peoplehr.com/](https://apidocs.peoplehr.com/)
[2][https://apidocs.peoplehr.com/#tag/Applicant%2Fpaths%2F~1Applicant%20%20-%20%20CreateNewApplicant%2Fpost](https://apidocs.peoplehr.com/#tag/Applicant%2Fpaths%2F~1Applicant%20%20-%20%20CreateNewApplicant%2Fpost)

### Related issues

_No response_

### PR DESCRIPTION
**Pull request by SanduDS**
Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/294

## Automation tests
 - Unit tests 
   > Passed
 - Integration tests
   > Passed

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes
 -

### CODE REVIEW

File: peoplehr/tests/test.bal [hunk 1]
```diff
@@ -186,4 +186,45 @@ function testGetAllVacancies() returns error? {
...
+    log:printInfo("Create new applicant");
+    NewApplicant newApplicant = {
+        FirstName: "Smith",
+        LastName: "Perera",
+        Email: "testapplicant@wso2.com",
+        Documents: [{DocumentName: "CV01", Url:"www.mycv.com"}],
+        Skills: "Programming",
+        VacancyReference: "VA100"
+    };
+    OperationStatus response= check baseClient->createNewApplicant(newApplicant);
```
Comment (LakshanSS) [inline peoplehr/tests/test.bal]:
```suggestion
    OperationStatus response = check baseClient->createNewApplicant(newApplicant);
```

Comment (LakshanSS) [reply peoplehr/tests/test.bal]: Check formatting in other places also.

File: peoplehr/types.bal [hunk 2]
```diff
@@ -818,3 +818,95 @@ public type HolidayDetail record {|
...
+public type AdditionalQuestion record {
+    string QuestionId;
+    string Value;
+};
+
+# Internal questions for create new applicant
+#
+public type InternalQuestion record {
+    *AdditionalQuestion;
+};
```
Comment (LakshanSS) [inline peoplehr/types.bal:835]:
```suggestion
};

```

File: peoplehr/types.bal [hunk 3]
```diff
@@ -818,3 +818,96 @@ public type HolidayDetail record {|
...
+|};
+
+# Applicant information for duplicate checking
+#
+# + VacancyReference - Vacancy reference for check duplicate applicant 
+# + FirstName - First name for check duplicate applicant(<= 50 characters) 
+# + LastName - Last name for check duplicate applicant(<= 50 characters) 
+# + Email - Email for check duplicate applicant
+public type ApplicantInformation record {|
+    string VacancyReference;
```
Comment (abeykoon) [inline peoplehr/types.bal]: Shall we make `VacancyReference ` 3 rd key of the record? We need to give priority to Applicant name.

**Review by abeykoon — APPROVED** [review]

**Comment by LakshanSS** [comment]
@SanduDS Please send same fix to `2201.0.0` branch also

**Pull request merged**